src/Npgsql/PreparedStatement.cs

```diff
@@ -95,12 +95,14 @@ internal void SetParamTypes(List<NpgsqlParameter> parameters)
     internal bool DoParametersMatch(List<NpgsqlParameter> parameters)
     {
         var paramTypes = ConverterParamTypes!;
-        var forall = paramTypes.Length == parameters.Count;
-        for (var i = 0; forall && i < paramTypes.Length; i++)
+        if (paramTypes.Length != parameters.Count)
```

Comment:
Kinda funny how we go back and forth

Reply:
I can't remember having changed it to this inferior forall, as for some reason initially I did have basically this code... Not sure what happened there.